core/src/main/java/org/elasticsearch/index/mapper/TokenCountFieldMapper.java

```diff
@@ -136,7 +136,7 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
 
         final int tokenCount;
         if (value == null) {
-            tokenCount = (Integer) fieldType().nullValue();
+            tokenCount = 0;
```

Comment:
@jimczi thanks, thats what I thought. @fred84 would you mind changing this and also adding another test that checks values that return a real `0` token count (e.g. empty Stings, or test where the analyzer doesn't return any tokens)?